feat: docker-compose with opensearch

yevhen-cap opened the pull request:
- core
- camunda platform

<img width="487" alt="image" src="https://github.com/user-attachments/assets/9e8a919a-68e1-4063-b9f0-8ea37eaf451e">
<img width="503" alt="image" src="https://github.com/user-attachments/assets/a1d6dc81-1382-4b73-b435-6d55f81aa823">
<img width="560" alt="image" src="https://github.com/user-attachments/assets/d595bf8c-5c52-46d6-985b-90739e97ea71">
<img width="374" alt="image" src="https://github.com/user-attachments/assets/ecbf1eca-747a-44b8-8191-41b8af4cebbe">

Commit:
chore: added docker-compose example with OpenSearch

- core
- camunda platform

Commit:
docker-compose with opensearch

- core
- camunda platform

Commit:
added missed new line

igpetrov commented on docker-compose-core-opensearch.yaml:
```diff
@@ -0,0 +1,149 @@
+      - operate
+
+  opensearch:
+    image: opensearchproject/opensearch:2.16.0
```
Can we move OS version to `.env` file?

igpetrov committed:
chore: renamed OS files

igpetrov committed:
chore: added OS test flows

igpetrov reviewed (approved):
LGTM

igpetrov commented:
Hi @yevhen-cap 

It seems like OpenSearch cluster never turns green and always stays yellow. Can you please take a look?

igpetrov commented on docker-compose.opensearch.yaml:
```diff
@@ -0,0 +1,400 @@
+
+services:
+
+  # opensearch-init:
```
Can you please uncomment or remove?

igpetrov commented on docker-compose.opensearch.yaml:
I see that ES version also has Kibana. Do we need to have Kibana too?

igpetrov commented:
> Hi @yevhen-cap
> 
> It seems like OpenSearch cluster never turns green and always stays yellow. Can you please take a look?

Likely related to this https://github.com/opensearch-project/opensearch-build/issues/4285

Commit:
Merge branch 'platform-opensearch' of github.com:camunda/camunda-platform into platform-opensearch

Commit:
Merge branch 'main' into platform-opensearch

Commit:
Merge branch 'platform-opensearch' of github.com:camunda/camunda-platform into platform-opensearch

Commit:
added recommended config

- health check workaround

aabouzaid reviewed (changes requested):
Adding a new entire Docker Compose file will increase the maintenance effort and duplication.
We need to do that via env var files instead of doing it with a new file.

Commit:
feat: opensearch improvement. unified compose files (#851)

* initial unified compose file

* removed ssl

* remove healthcheck

* removed test data

* fix: core changes

- added .env configs
- removed sh file
- unified docker-compose.yml and docker-compose-core.yml

* fix: core changes

- added recommended config
- removed ssl
- added .env configs
- removed sh file
- unified docker-compose.yml and docker-compose-core.yml

---------

Co-authored-by: Yevhen Podluzhnyi <kpiwelder@gmail.com>

igpetrov commented on .env:
```diff
@@ -48,3 +49,7 @@ RESOURCE_AUTHORIZATIONS_ENABLED=false
 OPENSEARCH_INITIAL_ADMIN_PASSWORD=OpenSearch_Admin1234!
+
+# search engine app: opensearch or elasticsearch
+XSEARCH=elasticsearch
+# XSEARCH=opensearch
```
Maybe consider giving more vocal name what the variable is doing, e.g.
```suggestion
SEARCH_DB=elasticsearch
# SEARCH_DB=opensearch
```

igpetrov commented on docker-compose-core.yaml:
```diff
@@ -116,9 +124,48 @@ services:
+      - "9601:9600"
+    restart: always
+    healthcheck:
+      test: [ "CMD-SHELL", "curl -k -f http://localhost:9200/_cluster/health -u admin:${OPENSEARCH_INITIAL_ADMIN_PASSWORD} | grep -q -E 'yellow|green'" ]
```
Suggesting to leave a note why yellow is okay-ish
```suggestion
      # Single node OpenSearch clusters are considered 'yellow', see https://github.com/opensearch-project/opensearch-build/issues/4285
      test: [ "CMD-SHELL", "curl -k -f http://localhost:9200/_cluster/health -u admin:${OPENSEARCH_INITIAL_ADMIN_PASSWORD} | grep -q -E 'yellow|green'" ]
```

yevhen-cap replied:
Agree with that suggestion

igpetrov reviewed (approved):
LGTM! Few optional comments.
Let's wait review/approval from @aabouzaid before we merge. Thank you!

Commit:
fix: edited README.md

- fixed comments
- added params for opensearch

Commit:
Merge branch 'main' into platform-opensearch

Commit:
use opensearch

Commit:
fix: fixed failed services

Commit:
fix: removed comments

aabouzaid reviewed (approved):
LGTM, thank you :+1:
